### ISSUE TITLE
Only one key observer permitted at a time?

### ISSUE THREAD

**Issue by bryan-vh** [hyperoslo/Cache#284]
Are we only allowed to have one observer for a key at a time? If I update a value in one of my VC's that then updates the observer in that VC but not in another VC that also has an observer for the same key. Is there a way to get multiple observers on a single key?

Thanks!

**Comment by 3lvis** [comment]
Hi @bryan-vh, I'm not sure, that's something related to Foundation and/or CoreData. 

Not sure if this is useful but was the first thing I found on the internet.
https://blog.pendowski.com/handling-multiple-observers-in-swift-2/

**Comment by bryan-vh** [comment]
Hi @3lvis, I ended up creating my own custom implementation that mirrors what the library does for key observers right now. As it stands, I believe that we can only have one observer for each key, since the map being used is [Key: (HybridStorage, KeyChange<Value>) -> Void]. In my implementation I changed this to [Key: [UUID: (Int) -> Void]], since my custom implementation is only for a counter that deals in Ints. Not sure if that's a good approach, but seems to work for me right now.
